app/docs/[...slug]/DocsPagesClient.tsx

```diff
@@ -45,6 +45,21 @@ export default function DocsClient({ props }) {
     title: DocumentationData?.next?.title,
   };
 
+  useEffect(() => {
+    const filepath = DocumentationData?.id;
+    const slug = filepath.substring(7, filepath.length - 4) + '/';
+    const recuseItems = (items) => {
+      items.forEach((item) => {
+        if (item.items) {
+          recuseItems(item.items);
+        } else if (item.slug === slug) {
+          setLearnActive(true);
+        }
+      });
+    };
+    recuseItems(NavigationLearnData?.data);
+  }, [NavigationLearnData, DocumentationData]);
```

Comment:
Made a few changes

1. There's a typo here in `recuseItems`, it looks like you're using recursion.
2. Updated useEffect to not run if `filepath` is null
3. I'm returning prematurely inside recurseItems if items is null (`NavigationLearnData?.data` is possibly null)

```suggestion
  useEffect(() => {
    const filepath = DocumentationData?.id;
    if(filePath){
    const slug = filepath.substring(7, filepath.length - 4) + '/';
    const recurseItems = (items) => {
      if(!items)
      {
      return
      }
      items.forEach((item) => {
        if (item.items) {
          recurseItems(item.items);
        } else if (item.slug === slug) {
          setLearnActive(true);
        }
      });
    };
    recurseItems(NavigationLearnData?.data);
    }
  }, [NavigationLearnData, DocumentationData]);
```